### PR TITLE
Add Endpoint for Creating Subscription Plans

### PR DESCRIPTION
**Pull request by iamaniekan**
## Description

This pull request introduces a new endpoint for managing subscription plans. Admins can now create various subscription plans with details including name, description, price, duration, and features. The changes include:

- **Endpoint Addition:** Added `POST /api/v1/plans` for creating new subscription plans.
- **Error Handling:** Implemented checks for duplicate plans and unauthorized access.
- **Database Models:** Updated the database schema to include a `SubscriptionPlan` model.
- **Testing:** Added pytest cases to ensure functionality and edge cases are covered.

---

## Related Issue (Link to issue ticket)

[Issue Ticket](https://github.com/hngprojects/hng_boilerplate_python_fastapi_web/issues/37)

---

## Motivation and Context

This feature is essential to allow admins to create and manage different subscription plans, which is critical for the platform's subscription-based service model. The addition of this endpoint ensures that various subscription options can be added and managed efficiently.

---

## How Has This Been Tested?

- **Unit Tests:** Added tests to validate the creation of subscription plans, handling of duplicate names, and authorization failures.
- **Integration Tests:** Verified end-to-end functionality using FastAPI's `TestClient`.
- **Test Environment:** Tests were run using a separate SQLite database to ensure no interference with the production environment.

Tests include:
- Creating a new subscription plan successfully.
- Handling attempts to create duplicate subscription plans.
- Responding correctly to unauthorized access.

---

### Bad Request
Bad Request

### Successful Request
Successful request

---

## Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to change)

---

## Checklist:

- [x] My code follows the code style of this project.
- [x] My change requires a change to the documentation.
- [x] I have updated the documentation accordingly.
- [x] I have read the **CONTRIBUTING** document.
- [x] I have added tests to cover my changes.
- [x] All new and existing tests passed.

---

### CODE REVIEW

File: api/v1/models/plans.py [hunk 1]
```diff
@@ -0,0 +1,33 @@
...
+    __tablename__ = "subscription_plans"
+    
+    id = Column(Integer, primary_key=True, index=True)
+    name = Column(String(255), nullable=False)
+    description = Column(Text, nullable=True)
+    price = Column(Integer, nullable=False)
+    duration = Column(String(50), nullable=False)
+    features = Column(ARRAY(String), nullable=False)
+    created_at = Column(DateTime, server_default=func.now(), nullable=False)
+    updated_at = Column(DateTime, server_default=func.now(), onupdate=func.now(), nullable=False)
```
Comment (Dev-wonderful) [inline api/v1/models/plans.py:33]: `id`, `created_at` and `updated_at` already exists in the `BaseModel` class

File: api/v1/routes/plans.py [hunk 2]
```diff
@@ -0,0 +1,22 @@
...
+from api.v1.models.plans import SubscriptionPlan
+from api.v1.schemas.plans import CreateSubscriptionPlan, SubscriptionPlanResponse
+from api.db.database import get_db
+from api.utils.dependencies import get_current_admin
+
+plans = APIRouter(prefix="/api/v1", tags=["Plans"])
+
+@plans.post("/plans", response_model=SubscriptionPlanResponse, status_code=status.HTTP_201_CREATED)
+def create_subscription_plan(plan: CreateSubscriptionPlan, db: Session = Depends(get_db), current_admin: str = Depends(get_current_admin)):
+    # Check to see if the plan already exist
```
Comment (Dev-wonderful) [inline api/v1/routes/plans.py:12]: Read the get current user tutorial I dropped in the channel, also the `current_admin` should be a `User` object, why are you using type `str`

**Review by Dev-wonderful — CHANGES_REQUESTED** [review]

File: api/v1/routes/plans.py (hunk 2 above)
Comment (iamaniekan) [reply api/v1/routes/plans.py]: I have resolved this following the Fast API documentation

File: api/v1/models/plans.py (hunk 1 above)
Comment (iamaniekan) [reply api/v1/models/plans.py]: I removed the highlighted field from the model.

**Review by Dev-wonderful — APPROVED** [review]

**Pull request merged**